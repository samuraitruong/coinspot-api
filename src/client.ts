import { BaseClient } from "./base-client";

export class CoinSpotClient extends BaseClient {
  constructor(apiKey: string, apiSecret: string) {
    super(apiKey, apiSecret, "https://www.coinspot.com.au/api");
  }

  async orders(cointype: string) {
    return this.postRequest("orders", { cointype });
  }

  async ordersHistory(cointype: string) {
    return this.postRequest("orders/history", { cointype });
  }

  async coinDeposit(cointype: string) {
    return this.postRequest("my/coin/deposit", { cointype });
  }

  async quoteSell(cointype: string, amount: number) {
    return this.postRequest("quote/sell", { cointype, amount });
  }

  async quoteBuy(cointype: string, amount: number) {
    return this.postRequest("quote/buy", { cointype, amount });
  }

  async myBalance() {
    return this.postRequest(`my/balances`, {});
  }

  async myOrders() {
    return this.postRequest("my/orders");
  }

  async myBuy(cointype: string, amount: number, rate: number) {
    return this.postRequest("my/buy", { cointype, amount, rate });
  }

  async mySell(cointype: string, amount: number, rate: number) {
    return this.postRequest("my/sell", { cointype, amount, rate });
  }

  async myBuyCancel(id: number) {
    return this.postRequest("my/buy/cancel", { id });
  }

  async mySellCancel(id: number) {
    return this.postRequest("my/sell/cancel", { id });
  }
}
