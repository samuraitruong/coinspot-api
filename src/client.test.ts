import { CoinSpotClient } from './client';
const client = new CoinSpotClient(process.env.API_TOKEN as string, process.env.API_SECRET as string);

describe("Client tests", () => {
  
 
  it.only("MyBalance test", async () => {
    const bl = await client.myBalance()
    expect(bl).toMatchSnapshot();
  })

  it.only("quoteSell test", async () => {
    const bl = await client.quoteSell('LUNA', 6959.62845592)
    expect(bl).toMatchSnapshot();
  })

  it.only("orderHistory test", async () => {
    const bl = await client.ordersHistory('RDD')
    expect(bl).toMatchSnapshot();
  })


})
