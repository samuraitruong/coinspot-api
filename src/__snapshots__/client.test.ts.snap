// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Client tests MyBalance test 1`] = `
Object {
  "balance": Object {
    "ae": 59.07626208,
    "aud": 20.440000000000005,
    "grs": 2.1434619,
    "iost": 109.95052226,
    "link": 0.55112815,
    "matic": 379.47783849,
    "rcn": 16.94886527,
    "rdd": 19362.83870967,
    "trx": 230.77633157,
    "vidt": 15.79899087,
    "xrp": 1,
  },
  "status": "ok",
}
`;

exports[`Client tests orderHistory test 1`] = `
Object {
  "orders": Array [],
  "status": "ok",
}
`;

exports[`Client tests quoteSell test 1`] = `
Object {
  "quote": null,
  "status": "ok",
  "timeframe": 4,
}
`;
